src/Main.hs

```diff
@@ -205,6 +206,45 @@ renderPage server page = with html_ [lang_ "en"] $ do
       let s = T.intercalate "|" $ T.replace " " "+" <$> fs
           url = "https://fonts.googleapis.com/css?family=" <> s <> "&display=swap"
        in stylesheet url
+    ogpMetaTags realm p = do
+      -- TODO: How to get base URL of the archive?
+      let baseUrl = "https://funprog.srid.ca/"
+          property_ = term "property"
+      meta_ [property_ "og:site_name", content_ $ realm <> " Archive"]
+      meta_ [property_ "og:url", content_ $ baseUrl <> pageUrl p]
+      case p of
+        Page_Index _ ->
+            meta_ [property_ "og:title", content_ "Home"]
+        Page_Stream (Rib.targetVal -> s) -> do
+            meta_ [property_ "og:title", content_ $ _streamName s]
+            meta_ [property_ "og:description", content_ $ _streamDescription s]
+        Page_Topic ((Rib.targetVal -> s), Rib.targetVal -> t) -> do
+            meta_ [property_ "og:title", content_ $ _topicName t]
+            meta_ [property_ "og:type", content_ "article"]
+            meta_ [property_ "og:article:section", content_ $ _streamName s]
+            -- TODO: Add a maybeProperty function?
+            meta_ [ property_ "og:article:modified_time"
+                  , content_ $ maybe "" ogpTimeFormat $ _topicLastUpdated t
+                  ]
+            maybe (return ())
+              (\msg -> do
+                  meta_
+                    [ property_ "og:article:published_time"
+                    , content_ $ ogpTimeFormat $ _messageTimestamp msg
+                    ]
+                  meta_
+                    [ property_ "og:description"
+                    , content_ $ T.take 300 $ _messageContent msg
+                    ]
+                  -- TODO: Is _messageAvatarUrl the full URL or just the path?
```

Comment:
It's the full URL. 

(Perhaps its type should be changed to `URI` to prevent this confusion in future).